### DISCUSSION TITLE
pnpm outdated does not yield expected results on pre-release dist tag packages

### DISCUSSION THREAD

**Discussion by cdaringe** [pnpm/pnpm#15790] [General]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

8.10.5

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Reproduction steps

**warning**: this isn't strictly a bug as much as it is undefined behavior.

- install a pre-release version, e.g. `depedencies: { "foo": "0.0.0-next-1" }`
- observe via `npm info foo` that the `next` dist tag's latest release is `0.0.0-next-2`
- run `pnpm outdated --json`, observe  that pnpm still thinks that latest release of the package on the `next` dist tag still reports `0.0.0-next-1`, not `0.0.0-next-2`

### Describe the Bug

pnpm outdated does not appear to consider from which dist-tag a package was published, yielding incorrect `outdated` output.

`npm` actually does seem to detect new releases on pre-releases.

### Expected Behavior

`pnpm outdated` produces correct outdated content, which shall consider the `dist-tag` that a package was sourced from.

It appears `pnpm outdated` only checks the implict `latest` npm dist-tag. This works for 99.9% of cases. However, I host a top level artifact (a server) for which I always wanting to be testing using the cutting edge of a dependency (which happens to publish on a pre-release dist-tag, e.g. `next`).

### Which Node.js version are you using?

20.8.0

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by Anoesj** [comment]
Same on Linux. Same without `--json` (so: `pnpm outdated`).

**Comment by cdaringe** [comment]
I ended up augmenting my internal greenkeeper-like product to run a post process that scans for these and runs update one by one. npm:semver is pretty sensitive to how it parses pre release versions if you try yourself. I had to convert my prerelease versions from #.#.#-TAG-#### to #.#.#-TAG.####, and it works nicely.

**Comment by yefrinoscar** [comment]
I'm working on a fix for this.

---
Written by an agent (Codex, GPT-5).
